cmp/zoneGrid/ZoneGridModel.ts

```diff
@@ -616,41 +625,65 @@ export class ZoneGridModel extends HoistModel {
         const ret = {} as Record<Zone, ZoneMapping[]>;
         forOwn(mappings, (rawMapping, zone) => {
             // 1) Standardize mapping into an array of ZoneMappings
-            const mapping = [];
+            let mapping = [];
             castArray(rawMapping).forEach(it => {
                 if (!it) return;
 
                 const ret = isString(it) ? {field: it} : it,
                     col = this.findColumnSpec(ret);
 
-                throwIf(!col, `Column not found for field ${ret.field}`);
-                return mapping.push(ret);
+                if (col) {
+                    mapping.push(ret);
+                } else {
+                    console.warn(`Column not found for field '${ret.field}'`);
+                }
             });
 
             // 2) Ensure mapping respects configured limits
             const limit = this.limits?.[zone];
             if (limit) {
-                throwIf(
-                    isFinite(limit.min) && mapping.length < limit.min,
-                    `Requires minimum ${limit.min} mappings in zone "${zone}"`
-                );
-                throwIf(
-                    isFinite(limit.max) && mapping.length > limit.max,
-                    `Exceeds maximum ${limit.max} mappings in zone "${zone}"`
-                );
+                if (isFinite(limit.min) && mapping.length < limit.min) {
+                    console.warn(
+                        `Requires minimum ${limit.min} mappings in zone "${zone}. Setting to default."`
+                    );
+                    mapping = this._defaultState.mappings[zone];
+                }
+
+                if (isFinite(limit.max) && mapping.length > limit.max) {
+                    const diff = mapping.length - limit.max;
+                    console.warn(
+                        `Exceeds maximum ${limit.max} mappings in zone "${zone}". Dropping last ${diff} fields`
+                    );
+                    mapping = dropRight(mapping, mapping.length - limit.max);
+                }
 
                 if (!isEmpty(limit.only)) {
+                    const offenders = [];
                     mapping.forEach(it => {
-                        throwIf(
-                            !limit.only.includes(it.field),
-                            `Field "${it.field}" not allowed in zone "${zone}"`
-                        );
+                        if (!limit.only.includes(it.field)) {
+                            console.warn(
+                                `Field "${it.field}" not allowed in zone "${zone}. Removing"`
+                            );
+                            offenders.push(it.field);
+                        }
                     });
+
+                    if (!isEmpty(offenders)) {
+                        remove(mapping, it => offenders.includes(it.field));
+                    }
+
+                    if ((zone == 'tl' || zone == 'tr') && isEmpty(mapping)) {
```

Comment:
Great catch! But I think we want to do this outside of the `!isEmpty(limit.only)` condition.

(I know enforcing `only` is currently the only place where we remove mappings, but I think the intent is clearer if we always do this as the final check of parseMappings)

Reply:
Great note Tom. I considered the same, but apparently made the wrong choice ;)